src/components/AutoComplete.tsx

```diff
@@ -0,0 +1,288 @@
+                            defaultChecked={item.value === value}
+                        />
+                        <Text size="s" color={gray7} as="span">
+                            {item.title}
+                        </Text>
+                    </StyledLabel>
+                ))}
+            </StyledRadioGroup>
+        </>
+    );
+}
+
+function getItemCreator<T>(onChange: (item: T) => void, map: React.MutableRefObject<AutoCompleteSelectedMap<T>>) {
+    return function createRenderItem<T1 extends T>(item: T1, index: number): AutoCompleteRenderItemProps<T> {
+        return {
+            item,
+            index,
+            checked: map.current.has(item),
+            onItemClick: () => onChange(item),
+        };
+    };
+}
+
+export function AutoCompleteList({ title, selected }: AutoCompleteListProps) {
+    const {
+        items,
+        value,
+        renderItem,
+        renderItems: Component,
+        renderState,
+        switchType,
+        popItem,
+        pushItem,
+        map,
+    } = useAutoCompleteContext();
+
+    useEffect(() => {
+        if (selected) {
+            switchType();
+        }
+    }, [selected, switchType]);
+
+    const onChange = useCallback(
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        (item: any) => {
+            const itemInMap = map.current.has(item);
+
+            if (itemInMap) {
+                popItem(item);
+            } else {
+                pushItem(item);
+            }
+        },
+        [map, popItem, pushItem],
+    );
+
+    const createRenderItem = getItemCreator(onChange, map);
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const itemsToRender: AutoCompleteRenderItemProps<any>[] = useMemo(() => {
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let target: Array<any>;
+
+        if (renderState === 'split') {
+            if (selected) {
+                target = value;
+            } else {
+                target = items.filter((item) => !map.current.has(item));
+            }
+        } else {
+            target = items;
+        }
+
+        return target.map(createRenderItem);
+    }, [renderState, items, value, selected, createRenderItem, map]);
+
+    return nullable(itemsToRender, (toRender) => (
+        <>
+            {nullable(title, (t) => (
+                <StyledText>{t}</StyledText>
+            ))}
+            <Component>{toRender.map(renderItem)}</Component>
+        </>
+    ));
+}
+
+export const AutoCompleteInput: React.FC<AutoCompleteInputProps> = ({ onChange, ...props }) => {
+    const handleInputChange = useCallback<React.ChangeEventHandler<HTMLInputElement>>((event) => {
+        onChange(event.target.value);
+    }, []);
+
+    return <Input {...props} onChange={handleInputChange} />;
+};
+
+const defaultRenderItems: React.FC<React.PropsWithChildren> = ({ children }) => <>{children}</>;
```

Comment:
?

Reply:
Inner default value for `renderItem` prop
What's confusing?